src/Eloquent/DocumentModel.php

```diff
@@ -0,0 +1,749 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Eloquent;
+
+use BackedEnum;
+use Carbon\CarbonInterface;
+use DateTimeInterface;
+use DateTimeZone;
+use Illuminate\Contracts\Queue\QueueableCollection;
+use Illuminate\Contracts\Queue\QueueableEntity;
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\Relation;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Exceptions\MathException;
+use Illuminate\Support\Facades\Date;
+use Illuminate\Support\Str;
+use MongoDB\BSON\Binary;
+use MongoDB\BSON\Decimal128;
+use MongoDB\BSON\ObjectID;
+use MongoDB\BSON\Type;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Laravel\Query\Builder as QueryBuilder;
+use Stringable;
+use ValueError;
+
+use function array_key_exists;
+use function array_keys;
+use function array_merge;
+use function array_unique;
+use function array_values;
+use function class_basename;
+use function count;
+use function date_default_timezone_get;
+use function explode;
+use function func_get_args;
+use function in_array;
+use function is_array;
+use function is_numeric;
+use function is_string;
+use function ltrim;
+use function method_exists;
+use function sprintf;
+use function str_contains;
+use function str_starts_with;
+use function strcmp;
+use function var_export;
+
+trait DocumentModel
+{
+    use HybridRelations;
+    use EmbedsRelations;
+
+    /**
+     * The parent relation instance.
+     */
+    private Relation $parentRelation;
```

Comment:
Changing the visibility to `private`. There is a getter and an setter for this method. It doesn't need to be protected. Since it's a trait, it can still be accessed directly in a class that uses the trait.